rs/boundary_node/rate_limits/canister/certification.rs

```diff
@@ -0,0 +1,174 @@
+use std::{cell::RefCell, rc::Rc};
+
+use ic_asset_certification::{Asset, AssetConfig, AssetEncoding, AssetFallbackConfig, AssetRouter};
+use ic_cdk::api::{data_certificate, set_certified_data};
+use ic_http_certification::{HeaderField, HttpCertificationTree, HttpRequest, HttpResponse};
+
+use crate::{
+    metrics::{export_metrics_as_http_response, with_metrics_registry},
+    state::with_canister_state,
+};
+
+thread_local! {
+    static HTTP_TREE: Rc<RefCell<HttpCertificationTree>> = Default::default();
+
+    // initializing the asset router with an HTTP certification tree is optional.
+    // if direct access to the HTTP certification tree is not needed for certifying
+    // requests and responses outside of the asset router, then this step can be skipped.
+    static ASSET_ROUTER: RefCell<AssetRouter<'static>> = RefCell::new(AssetRouter::with_tree(HTTP_TREE.with(|tree| tree.clone())));
+}
+
+// Handlers
+pub fn serve_asset(request: &HttpRequest<'static>) -> HttpResponse<'static> {
+    let path = match request.get_path() {
+        Ok(path) => path,
+        Err(err) => {
+            return HttpResponse::builder()
```

Comment:
This response will fail response verification since it's not certified. A potential solution could be to add a `/400-malformed-path` path and if extraction of the path fails, then serve the 400 path instead. This would require the response body to be fixed though.

Reply:
Scratch this, it won't work. I need to think about this more.

Reply:
could You please make a commit directly, cause i'm not sure i will do it correctly myself